Subject: [PATCH] Gallery2: Fix thumbnails of new edit photos display wrongly in Documents.

The photo is edited and save to new file in gallery,
the thumbnail of the new photo is generated and saved at the same time.
But the directory of thumbnails is not created, and the thumbnail can't save.
So the thumbnail displays incorrectly in Documents's Recent list.

Check the directory of thumbnails whether create,
if not create it at first to make the thumbnails save successfully.

diff --git a/src/com/android/providers/media/MediaProvider.java b/src/com/android/providers/media/MediaProvider.java
index e181e59..c2fbb74 100644
--- a/src/com/android/providers/media/MediaProvider.java
+++ b/src/com/android/providers/media/MediaProvider.java
@@ -5005,6 +5005,14 @@ private File queryForDataFile(Uri uri) throws FileNotFoundException {
                         if (data == null) {
                             throw new FileNotFoundException("Null path for " + uri);
                         }
+                        File directory = new File(data).getParentFile();
+                        if (!directory.isDirectory()) {
+                            if (!directory.mkdirs()) {
+                                Log.e(TAG, "Unable to create .thumbnails directory for " + data);
+                                throw new FileNotFoundException(
+                                        "Unable to create .thumbnails directory " + uri);
+                            }
+                        }
                         return new File(data);
                     } else {
                         throw new FileNotFoundException("Unable to read entry for " + uri);
